internal/pkg/controllers/profile/profile.go

```diff
@@ -116,11 +119,13 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 		profilePath, err := getProfilePath(name, profile)
 		if err != nil {
 			logger.Error(err, "cannot get profile path")
+			r.record.Event(profile, event.Warning(event.Reason("cannot get profile path"), err))
```

Comment:
What's your thoughts on adding this into a new implementation of `logger`?

Reply:
And having a combined logger / eventer? So it would look something like `logger.Event()`?

Reply:
Just noticed the passed in logger is no longer being used

Reply:
Logging an error automatically throws an event? Sound good! :)

Reply:
Hmm I'm not sure we will always want to have our logging and eventing tied together in that manner. @pjbgf did you have a specific use case in mind?

Reply:
The idea was just to remove repetition as I can easily imagine those two lines being used quite frequently together.
I thought more like an EventLogger that implemented the same interface as logger but had an extra side effect.
Thinking a bit more about it, this would violate SRP and probably would make things a bit more complex for new contributors to understand.

Let's keep it simple for now, if needs be we can always review and refactor. ;) 

/lgtm